### PR TITLE
docs: Groom the documentation

### PR DESCRIPTION
**Pull request by darinspivey**
Some of the technical docs needed to be cleaned up in terms of
how we show parameter types and default values. Various other
small corrections, too.

### CODE REVIEW

File: docs/index.md [hunk 1]
```diff
@@ -16,27 +16,28 @@ resource "logdna_view" "http500" {
...
-- If you do not provide a specific base URL in the provider configuration, the base url defaults to `https://api.logdna.com`.
-- If you want to create an Alert that uses PagerDuty to notify you, you will need to provide LogDNA with the [PagerDuty API key](https://support.pagerduty.com/docs/generating-api-keys#events-api-keys). To ensure that the LogDNA Dashboard properly displays the PagerDuty alert notification channel, we recommend that you first link the PagerDuty service to LogDNA via the [Dashboard UI](https://docs.logdna.com/docs/pagerduty-alert-integration), before using the Configuration API to create a PagerDuty Alert. However, not doing so doesn't in any way prevent the use of the Configuration API to create Alerts that use PagerDuty.
+- Authentication is handled via the `servicekey` parameter and can be set in the `provider` configuration section in the `.tf` file.
+- Be aware that the underlying LogDNA Configuration API has a rate limit of 50 requests per minute. Therefore, when using the LogDNA Terraform provider, there is also a limit of 50 resource operations per minute.
```
Comment (darinspivey) [inline docs/index.md:34]: These 2 sentences are redundant and say the same thing. Remove the second one, or make that one the primary sentence (since it talks about "resource operations" rather than requests)

File: examples/many_views.tf [hunk 2]
```diff
@@ -2,103 +2,22 @@ provider "logdna" {
...
+  name  = "not_info_logs"
+  query = "-level:info"
 }
 
-resource "logdna_view" "my_view4" {
-  name  = "test4"
-  query = "test"
+resource "logdna_view" "exception_errors" {
+  name  = "exception_errors"
+  query = "ClassCastException OR NullPointerException"
```
Comment (darinspivey) [inline examples/many_views.tf:22]: Some of these examples should use more properties than just name/query.  It's a little redundant otherwise.

**Comment by darinspivey** [comment]
Looks like there could be a few more examples for alerts, or at a minimum, stop using `test@logdna.com`.  That's a real address and it should be indicative of the user's email.

File: docs/index.md [hunk 3]
```diff
@@ -16,27 +16,28 @@ resource "logdna_view" "http500" {
...
+- If you do not provide a specific a `url` in the provider configuration, the URL defaults to `https://api.logdna.com` (recommended).
+- If you want to create an Alert that uses PagerDuty to notify you, you will need to provide LogDNA with the [PagerDuty API key](https://support.pagerduty.com/docs/generating-api-keys#events-api-keys). To ensure that the LogDNA Dashboard properly displays the PagerDuty alert notification channel, we recommend that you first link the PagerDuty service to LogDNA via the [Dashboard UI](https://docs.logdna.com/docs/pagerduty-alert-integration) before using this plugin to create a PagerDuty Alert. However, not doing so doesn't in any way prevent this plugin from creating such resources (they just won't work until the linkage is correct).
```
Comment (david-ly) [inline docs/index.md:36]: > However, not doing so doesn't in any way prevent this plugin from creating such resources (they just won't work until the linkage is correct).

Nit: this sentence structure sounds kinda strange, I might reword it to something like `You may choose to create such resources first and then link PagerDuty, but be aware that they will not work as intended until the connection is reconciled.`

**Review by david-ly — DISMISSED** [review]
A few non-blocking nits since this PR is just intended for grooming, but I do think there are some overall improvements and changes we should make for the docs in the future.

File: docs/index.md [hunk 4]
```diff
@@ -16,27 +16,28 @@ resource "logdna_view" "http500" {
...
-- Be aware that the underlying LogDNA Configuration API has a rate limit of 50 requests per minute; therefore, when using the LogDNA Terraform provider, there is also a limit of 50 resource operations per minute.
-- If you do not provide a specific base URL in the provider configuration, the base url defaults to `https://api.logdna.com`.
-- If you want to create an Alert that uses PagerDuty to notify you, you will need to provide LogDNA with the [PagerDuty API key](https://support.pagerduty.com/docs/generating-api-keys#events-api-keys). To ensure that the LogDNA Dashboard properly displays the PagerDuty alert notification channel, we recommend that you first link the PagerDuty service to LogDNA via the [Dashboard UI](https://docs.logdna.com/docs/pagerduty-alert-integration), before using the Configuration API to create a PagerDuty Alert. However, not doing so doesn't in any way prevent the use of the Configuration API to create Alerts that use PagerDuty.
+- Authentication is handled via the `servicekey` parameter and can be set in the `provider` configuration section in the `.tf` file.
```
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/index.md:33]: The goal here is to point out that these are things for the `.tf` file.  There are other files like `.terraformrc` which have different configurations not related to this.  I think this clarifies.  In other words: "Where is the provider configuration?".  Answer: in the `.tf` file.

File: docs/index.md [hunk 5]
```diff
@@ -16,27 +16,28 @@ resource "logdna_view" "http500" {
...
+- Be aware that the underlying LogDNA Configuration API has a rate limit of 50 requests per minute. Therefore, when using the LogDNA Terraform provider, there is also a limit of 50 resource operations per minute.
+- If you do not provide a specific a `url` in the provider configuration, the URL defaults to `https://api.logdna.com` (recommended).
+- If you want to create an Alert that uses PagerDuty to notify you, you will need to provide LogDNA with the [PagerDuty API key](https://support.pagerduty.com/docs/generating-api-keys#events-api-keys). To ensure that the LogDNA Dashboard properly displays the PagerDuty alert notification channel, we recommend that you first link the PagerDuty service to LogDNA via the [Dashboard UI](https://docs.logdna.com/docs/pagerduty-alert-integration) before using this plugin to create a PagerDuty Alert. However, not doing so doesn't in any way prevent this plugin from creating such resources (they just won't work until the linkage is correct).
 
 ## Argument Reference
 
-The following arguments are supported:
+The following arguments are contained in the `provider` section of the `.tf` file:
```
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/index.md:40]: Same reason here--specify that it's the `.tf` file, at least for the provider-specific one.  I've made the others into the format:
```
The following arguments are supported by \`logdna_alert\`:
```

File: docs/resources/logdna_alert.md [hunk 6]
```diff
@@ -75,45 +75,45 @@ resource "logdna_alert" "my_alert" {
...
-- `timezone`: _(Optional)_ Which time zone the log timestamps will be formatted in. Timezones are represented as [database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones), type _string_
-- `triggerinterval`: _(Optional)_ Interval which the Alert will be looking for presence or absence of log lines. For presence Alerts, valid options are: `30`, `1m`, `5m`, `15m`, `30m`, `1h`, `6h`, `12h`, and `24h`. For absence Alerts, valid options are: `15m`, `30m`, `1h`, `6h`, `12h`, and `24h`. Type _string_ (**Defaults: "30" for presence; "15m" for absence**)
-- `triggerlimit`: _(Required)_ Number of lines before the Alert is triggered. (eg. Setting a value of `10` for an `absence` Alert would alert you if `10` lines were not seen in the `triggerinterval`), type _integer_
+- `emails`: **_[]string (Required)_** An array of email addresses (strings) to notify in the Alert
+- `immediate`: **_string_** _(Optional; Default: `"false"`)_ Whether the Alert will trigger immediately after the trigger limit is reached. _Note: Immediate can only be set to `"true"` for presence alerts_. Valid options are `"true"` and `"false"` for presence Alerts and `"false"` for absence Alerts.
```
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md:87]: ugh, this has to match with the API, and I'm not sure of the rules there.  It's also worth noting that I just hit the problem where one of `immediate` or `terminal` has to be `true` (at least for webhooks), but I see no way to validate them both ahead of time.

We have to check the API functionality before this can be confirmed.

File: docs/resources/logdna_alert.md [hunk 7]
```diff
@@ -75,45 +75,45 @@ resource "logdna_alert" "my_alert" {
...
+- `operator`: **_string_** _(Optional; Default: `presence`)_ Whether the Alert will trigger on the presence or absence of logs. Valid options are `presence` and `absence`.
+- `terminal`: **_string_** _(Optional; Default: `"true"`)_ Whether the Alert will trigger after the `triggerinterval` if the Alert condition is met (e.g. send an Alert after 30s). Valid options are `"true"` and `"false"` for presence Alerts and `"true"` for absence Alerts.
+- `timezone`: **_string_** _(Optional; Defaults: `"30"` for presence, `"15m"` for absence)_ Which time zone the log timestamps will be formatted in. Timezones are represented as [database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones).
```
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md:90]: I believe I checked this, yes.  This is copy/pasted from the other, but we need to verify.

File: docs/resources/logdna_alert.md [hunk 8]
```diff
@@ -75,45 +75,45 @@ resource "logdna_alert" "my_alert" {
...
+- `triggerinterval`: **_string_** _(Optional; Defaults: `"30"` for presence; `"15m"` for absence)_ Interval which the Alert will be looking for presence or absence of log lines. For presence Alerts, valid options are: `30`, `1m`, `5m`, `15m`, `30m`, `1h`, `6h`, `12h`, and `24h`. For absence Alerts, valid options are: `15m`, `30m`, `1h`, `6h`, `12h`, and `24h`.
```
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md:91]: YESSSSS, OMG, it's bad.  There's no fixing that.  What a mistake we made!

File: docs/resources/logdna_view.md [hunk 9]
```diff
@@ -74,47 +74,47 @@ The following arguments are supported:
...
 _Note:_ A `name` and at least one of the following properties: `apps`, `hosts`, `levels`, `query`, `tags` must be specified to create a View.
 
-- `apps`: _(Optional)_ Array of names of apps (each app is of type _string_) to filter the View by
-- `categories`: _(Optional)_ Array of existing category names (each category is of type _string_) this View should be nested under. _Note: If the category does not exist, the View will by default be created in uncategorized_
-- `hosts`: _(Optional)_ Array of names of hosts (each host is of type _string_) to filter the View by
-- `levels`: _(Optional)_ Array of names of levels (each level is of type _string_) to filter the View by
-- `name`: _(Required)_ Name this View will be given, type _string_
-- `query`: _(Optional)_  Search query scope for the View, type _string_
-- `tags`: _(Optional)_ Array of names of tags (each tag is of type _string_) to filter the View by
+- `apps`: **_string_** _(Optional)_ Array of names of apps (each app is of type _string_) to filter the View by.
```
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_view.md:77]: Can you point me to a link and specific suggestion?  I'm not sure what you're referring to.

File: docs/index.md (hunk 1 above)
Comment (darinspivey) [reply docs/index.md]: fixed

File: examples/many_views.tf (hunk 2 above)
Comment (darinspivey) [reply examples/many_views.tf]: fixed

File: docs/resources/logdna_alert.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (david-ly) [reply docs/resources/logdna_alert.md:87]: Yeah makes sense, this might just be something we'll have to flag for now and come back to later once we know more

File: docs/resources/logdna_view.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (david-ly) [reply docs/resources/logdna_view.md:77]: Yeah, `hosts` for example:
> - `hosts`: **[]string** _(Optional)_ Array of host names to filter the View by.

For consistency and less redundancy (imo), we can use this same pattern for `apps`: 
> - `apps`: **[]string** _(Optional)_ Array of app names to filter the View by.

File: docs/resources/logdna_alert.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (david-ly) [reply docs/resources/logdna_alert.md:90]: Yeah it looks like these defaults were copy/pasted over from `triggerinterval` but they don't seem correct in the context of the `timezone` argument here

[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md:90]: ah, I probably muffed that up. Will check.

[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md:90]: Confirmed my mistake.  Good catch!

File: docs/resources/logdna_alert.md (hunk 6 above)
[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md:87]: Ok, it's a wording thing.  It tries to say that `true` AND `false` are valid options for `presence`, but only `false` is available for absence alerts.  Removing that first part should clear it up.

[the remark this answers is not in the document]
Comment (darinspivey) [reply docs/resources/logdna_alert.md]: fixed, good catch

**Review by mdeltito — APPROVED** [review]
✨ very nice!
